spec/acts_as_tenant/model_extensions_spec.rb

```diff
@@ -461,4 +463,98 @@
       expect(ActsAsTenant.current_tenant).to eq(@account)
     end
   end
+
+  describe 'overlapping tenants' do
+    before(:each) do
+      ActsAsTenant.without_tenant do
+        @account1 = Account.create!
+        @account2 = Account.create!
+        @project1 = Project.create!(name: 'Project1')
+        @project2 = Project.create!(name: 'Project2')
+      end
+
+      ActsAsTenant.with_tenant(@account1) do
+        @manager1 = Manager.create!(name: 'Manager1')
+      end
+
+      ActsAsTenant.with_tenant(@project1) do
+        @manager2 = Manager.create!(account: @account1, name: 'Manager2')
+        @manager1.update!(project_id: @project1.id)

```

Comment:
couldn't you just do `@manager1.save!` (without explicitly setting `project_id`) and have it update the `:project` tenant?

Reply:
Sure, but this way it's a little more explicit which helps communicate the purpose of this test better

Reply:
oh okay I was more so asking if performing actions on records would be enough to automatically set the tenant. but I guess the first line technically already does that

Reply:
tenant will automatically be set as soon as you validate the model, so yes, calling `.save` will automatically do it